### PR TITLE
Use QuoteStrategy from Configuration to get table names

### PR DESCRIPTION
**Pull request by plfort**
Use QuoteStrategy to get table names
See #191

### CODE REVIEW

**Comment by plfort** [comment]
I didn't test on 2.4.* ...
I will fix it

**Comment by alexsegura** [comment]
@plfort wow your PR is way better than mine https://github.com/doctrine/data-fixtures/pull/204

Did you manage to fix the tests?

**Comment by fon60** [comment]
I think there was some error on travis, because only PHP5.3 failed. So this must have been some kind of glitch or error in Anotations regarding this version of PHP or uncleared cache. Can someone rerun tests on travis?

**Comment by alexsegura** [comment]
@fon60 actually it failed with PHP 5.3 & `doctrine-orm` 2.4.7, look at the output of Composer in Travis

This is because `doctrine-orm` 2.5.* needs at least PHP 5.4, while `doctrine-orm` 2.4.7 works with PHP 5.3

**Comment by alexsegura** [comment]
@plfort  the Travis build completed successfully, great :smile: 

@lavoiesl do you think we can merge this PR ?

**Comment by edigu** [comment]
+1 on this.

I hit the same problem today, ORMPurger ignores schema name on postgresql and produces wrong query when running fixtures.

 - Expected query to purge : `DELETE FROM core.users`
 - Actual query : `DELETE FROM users`

My entity:

```php
<?php
namespace Core\Entity;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity
 * @ORM\Table(name="users", schema="core")
 */
class User
{
}
```

and output of the `data-fixture:import` command is:

```
[Doctrine\DBAL\Exception\TableNotFoundException]                             
  An exception occurred while executing 'DELETE FROM users':                   
  SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "users" does not exist  
```

**Comment by alexsegura** [comment]
@edigu did you try with this branch?

**Comment by edigu** [comment]
Yes. I just tried now with `dev-master#7c4d7c` and my fixtures are loaded without any problems event with `--purge-with-truncate` argument. So i can confirm that ORMPurger properly generates the delete query with this PR. @plfort thanks for your effort.

**Comment by fon60** [comment]
So now what? I'm waiting for this PR very much.

**Comment by alexsegura** [comment]
Not sure who are the maintainers of this project

**Comment by guilhermeblanco** [comment]
:+1:

File: lib/Doctrine/Common/DataFixtures/Purger/ORMPurger.php [hunk 1]
```diff
@@ -198,15 +199,42 @@ private function getAssociationTables(array $classes, AbstractPlatform $platform
...
+    }
+
+    /**
+     *
+     * @param array            $association
+     * @param \Doctrine\ORM\Mapping\ClassMetadata    $class
+     * @param \Doctrine\DBAL\Platforms\AbstractPlatform $platform
+     * @return string
+     */
+    private function getJoinTableName($assoc, $class, $platform)
```
Comment (stof) [inline lib/Doctrine/Common/DataFixtures/Purger/ORMPurger.php]: typehints should be added

File: tests/Doctrine/Tests/Common/DataFixtures/Purger/ORMPurgerTest.php [hunk 2]
```diff
@@ -34,11 +39,30 @@ public function testGetAssociationTablesQuoted()
...
         $class = new ReflectionClass('Doctrine\Common\DataFixtures\Purger\ORMPurger');
         $method = $class->getMethod('getAssociationTables');
         $method->setAccessible(true);
         $associationTables = $method->invokeArgs($purger, array(array($metadata), $platform));
         $this->assertEquals($associationTables[0], '"INSERT"');
     }
+
+    public function testTableNameWithSchema()
+    {
+        $isDoctrine25 = (ORMVersion::compare('2.5.0') <= 0);
```
Comment (stof) [inline tests/Doctrine/Tests/Common/DataFixtures/Purger/ORMPurgerTest.php]: braces are not needed

**Comment by FractalizeR** [comment]
Will this make into 1.x?

**Comment by fon60** [comment]
I'm also waiting for release with this fix.

**Comment by FractalizeR** [comment]
For now I'm just doing all manually with --append ;)

**Comment by awildeep** [comment]
I am still having issues with this.  Can anyone send me a working example of a composer.json using symfony3?

No matter what I try I can't get the fixtures to detect my schema names for my entities.  It detects my schema names on my ManyToMany join tables, but not my actual entities.

**Comment by awildeep** [comment]
So I found my issue, but this took way longer to resolve than I expected:

my composer.json had:
"doctrine/doctrine-fixtures-bundle": "^2.3"

But did not contain a reference to "doctrine/data-fixtures".  As such I was grabbing v1.1.1 of "doctrine/data-fixtures", which is the latest stable release however it does NOT include this correction.

Anyone care to comment on when we might be getting an updated stable release?

To correct my problem I had to add this to my composer.json:
"doctrine/data-fixtures": "dev-master"

**Pull request merged**